### PR TITLE
Modernize logging and code style across the codebase

### PR DESCRIPTION
**Pull request by joonson**
## Summary
- Replace `print()`/`scorefile.write()` dual logging with Python's `logging` module, writing to both stdout and `scores.txt` via handlers
- Add `--gpu_id` argument for single-GPU training, setting `CUDA_VISIBLE_DEVICES` so any physical GPU maps to logical GPU 0
- Guard all logging behind `gpu == 0` to prevent duplicate output in distributed training
- Use `tqdm` progress bar in `train_network` (with `disable=not verbose` for multi-GPU)
- Convert all `%`-style and `.format()` string formatting to f-strings, preserving type specifiers (`:d`, `:f`, etc.) for safety
- Remove unused imports (`pdb`, `time`, `threading`, `numpy`, `DataLoader`, `wavfile`) across 11 files
- Remove trailing semicolons across 8 files
- Fix `== None` → `is None`
- Add `.DS_Store` and `Thumbs.db` to `.gitignore`

### CODE REVIEW

**Comment by joonson** [comment]
## Manual Test Checklist

### 1. Fresh training (single GPU)
```bash
python ./trainSpeakerNet.py --config ./configs/ResNetSE34L_AP.yaml
```
- [x] Training starts without errors
- [x] tqdm shows a proper progress bar with percentage from epoch 1
- [x] Loss decreases over a few epochs
- [x] Checkpoint saved at `test_interval`
- [x] Log file created at `exps/ResNetSE34L_AP/result/scores.txt`

### 2. Resume from new checkpoint
```bash
# Stop training, then restart with same command
python ./trainSpeakerNet.py --config ./configs/ResNetSE34L_AP.yaml
```
- [x] Loads the latest checkpoint and prints "loaded from previous state"
- [x] Resumes at the correct epoch (not epoch 1)
- [x] Learning rate is correct (optimizer/scheduler restored from checkpoint)

### 3. Load a legacy checkpoint (backward compatibility)
```bash
# Use an old .model file from before the code changes
python ./trainSpeakerNet.py --eval --model ResNetSE34L --log_input True --trainfunc angleproto --save_path exps/test --eval_frames 400 --initial_model baseline_lite_ap.model
```
- [x] Loads without errors
- [ ] EER matches expected `2.1792`
- [x] tqdm shows extraction and scoring progress bars

### 4. Load a legacy `{"model": state_dict}` checkpoint
If you have any old checkpoint saved in that format:
- [x] Loads without errors
- [x] Keys are remapped (`__S__.` prefix added)

### 5. Eval with pretrained ResNetSE34V2
```bash
python ./trainSpeakerNet.py --eval --model ResNetSE34V2 --log_input True --encoder_type ASP --n_mels 64 --trainfunc softmaxproto --save_path exps/test --eval_frames 400 --initial_model baseline_v2_smproto.model
```
- [x] Loads without errors
- [x] EER matches expected `1.0180`

### 6. Eval with pretrained RawNet3
```bash
python ./trainSpeakerNet.py --eval --config ./configs/RawNet3_AAM.yaml --initial_model models/weights/RawNet3/model.pt
```
- [x] Loads without errors
- [x] EER matches expected `0.8932`

### 7. Multi-GPU distributed training
```bash
export CUDA_VISIBLE_DEVICES=0,1
python ./trainSpeakerNet.py --config ./configs/ResNetSE34L_AP.yaml --distributed
```
- [x] All GPUs initialize without errors
- [x] Only rank 0 shows tqdm progress bar
- [x] Only rank 0 writes to log file
- [x] Training completes a few epochs

### 8. Mixed precision training
```bash
python ./trainSpeakerNet.py --config ./configs/ResNetSE34L_AP.yaml --mixedprec
```
- [x] No errors from `autocast` or `GradScaler`
- [x] Loss values are comparable to FP32 training

### 9. Training with augmentation
```bash
python ./trainSpeakerNet.py --config ./configs/RawNet3_AAM.yaml
```
(RawNet3 config has `augment: True`)
- [x] Augmentation paths resolve correctly (pathlib)
- [x] No errors from noise type extraction (`file.parts[-4]`)

### 10. Data preparation
```bash
python ./dataprep.py --save_path data --convert
```
- [x] tqdm progress bar works for conversion
- [x] No regressions in dataprep.py
